feat: add GET /outgoing-payment-grant endpoint

BlairCurrey opened the pull request:
- add `GET /outgoing-payment-grant` endpoint
- remove identity field on outgoing payment access requirement. This will allow introspection with or without the identifier as discussed in slack here https://interledgerfoundation.slack.com/archives/C06KZCKEEKW/p1763493339094139?thread_ts=1763480687.558089&cid=C06KZCKEEKW

fixes https://github.com/interledger/open-payments-specifications/issues/15
will be used to completed https://github.com/interledger/rafiki/issues/3375

BlairCurrey committed:
feat: add outgoing-payment-grant endpoint

and remove required identity field on outgoing payment access

BlairCurrey committed:
chore: update version

BlairCurrey commented on openapi/resource-server.yaml:
```diff
@@ -800,6 +800,47 @@ paths:
+                        $ref: '#/components/schemas/amount'
+                    required:
+                      - spentReceiveAmount
+                      - spentDebitAmount
```
What should we return when there is no outgoing payment grant spent amount? Would happen if the grant has been created but no payments yet, for example. Or maybe if there is an interval and no payments have been made in it yet.

My instinct was that simply constitutes 0. However, what will the asset scale and code be? We dont know without the spent amount. Can we reliably get what it will be somewhere else? Im not sure we can. Maybe we could get it off the quote but the quote is optional.

So alternatives I see would be changing this to not be required (null or undefined then client checks that before doing any incrementing, number comparison, etc.). Or make the asset scale and code optional for that case. Or values like an empty string, 0 ... but yuck.

mkurapov replied:
Having these fields as optional would be the most straightforward. Or, if we want the amounts before any payments have been created, then we can use the `limits` object in the outgoing payment access of the token introspection response

BlairCurrey replied:
I will change them to optional but also look into using the limits when implementing in rafiki. I suppose maybe even if we want to use limits we may not always have them (a grant without spent or receive limits? - edge case because why would you look them up, but still)

mkurapov replied:
> we may not always have them

Actually yes, that is true. For outgoing payment limits in a grant, they are `debitAmount` XOR `receiveAmount`. So we won't be able to get both assets for spent amounts lookup before any payments are created.

mkurapov commented on openapi/auth-server.yaml:
```diff
@@ -518,7 +518,6 @@ components:
       required:
         - type
         - actions
-        - identifier
```
I think in the grant request we can still keep this as a required, no?

Token introspection, we will just have a unique case where the `identifier` is not required when calling this route

BlairCurrey replied:
I will remove this, after digging into it a bit more. 

I was doing this because the token introspection request was failing at the validator. This is because on my (old) branch the token introspection open api spec references this auth-server spec. on Rafiki `main` I see we are now inlining all that stuff, so I can just update the token introspection spec in Rafiki (after updating my branch 😬) . I also see the point that this change has more implications than just loosening token introspection.

mkurapov commented on openapi/auth-server.yaml:
```diff
@@ -1,7 +1,7 @@
 info:
   title: Open Payments Authorization Server
-  version: '1.1.1'
+  version: '1.2.1'
```
```suggestion
  version: '1.2.0'
```

BlairCurrey replied:
fixed all [e107895](https://github.com/interledger/open-payments-specifications/pull/24/commits/e107895936d7dca992bd6dca2ff7ff2087a6bb07)

BlairCurrey committed:
fix: return identifier to access op access item

https://github.com/interledger/open-payments-specifications/pull/24/files#r2546989934

BlairCurrey committed:
fix: version

BlairCurrey committed:
chore: rm amounts as required

https://github.com/interledger/open-payments-specifications/pull/24#discussion_r2546893832

BlairCurrey committed:
fix: maybe fix lint error

BlairCurrey committed:
fix: maybe fix lint error

mkurapov commented on openapi/resource-server.yaml:
```diff
@@ -800,6 +800,44 @@ paths:
+        '403':
+          $ref: '#/components/responses/403'
+      description: |-
+        Returns the spent amounts for the current grant corresponding to the presented GNAP access token, if any. If a grant was created with an interval (recurring), the amounts returned are for the current interval.
```
```suggestion
        Returns the spent amounts for the current outgoing payment grant corresponding to the presented GNAP access token, if any. If a grant was created with an interval (recurring), the amounts returned are for the current interval.
```

BlairCurrey committed:
Update openapi/resource-server.yaml

Co-authored-by: Max Kurapov <max@interledger.org>
